language/vm/vm-derive/src/lib.rs

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) The Libra Core Contributors
```

Comment:
It seems bizarre to me that this is in 'vm'

I feel like the VM shouldn't know about the IR at all

Reply:
Same to me. Although I kinda understand it could be because of the way proc macros derivations work.

Reply:
Yeah, we play a two-step trick to make this happen.

(FWIW "vm" is not a great name for this crate -- it's more like "vm-types".)